lib/HttpClient.js

```diff
@@ -105,6 +107,16 @@ function RequestTimeoutError(ms) {
 }
 util.inherits(RequestTimeoutError, Error);
 
+function TooManyRedirectsError(redirectNumber) {
+    if (Error.captureStackTrace) {
+        Error.captureStackTrace(this, TooManyRedirectsError);
+    }
+
+    this.message = 'Aborted after ' + redirectNumber + ' redirects';
+    this.name = 'TooManyRedirectsError';
+}
+util.inherits(TooManyRedirectsError, Error);
```

Comment:
Yes, we should definitely use `makeConstructor` for the new stuff. Don't worry about the existing constructors (unless you want to give it a shot!).